examples/Cassie/run_dircon_squatting.cc

```diff
@@ -0,0 +1,817 @@
+class BodyPonitPositionConstraint : public DirconAbstractConstraint<double> {
+ public:
+  BodyPonitPositionConstraint(const RigidBodyTree<double>& tree,
+                              string body_name,
+                              Vector3d translation,
+                              Vector3d desired_pos) :
+    DirconAbstractConstraint<double>(3,
+                                     tree.get_num_positions(),
+                                     VectorXd::Zero(3),
+                                     VectorXd::Zero(3),
+                                     body_name + "_position_constraint"),
+    tree_(tree),
+    body_idx_(multibody::GetBodyIndexFromName(tree, body_name)),
+    translation_(translation),
+    desired_pos_(desired_pos) {
+  }
+  ~BodyPonitPositionConstraint() override = default;
+
+  void EvaluateConstraint(const Eigen::Ref<const drake::VectorX<double>>& x,
+                          drake::VectorX<double>* y) const override {
+    VectorXd q = x;
+    KinematicsCache<double> cache = tree_.doKinematics(q);
+    VectorXd pt = tree_.transformPoints(cache, translation_, body_idx_, 0);
+
+    *y = pt - desired_pos_;
+  };
+ private:
+  const RigidBodyTree<double>& tree_;
+  const int body_idx_;
+  const Vector3d translation_;
+  const Vector3d desired_pos_;
+};
+
+// Use fixed-point solver to get configuration guess (for standing in place)
+void GetInitFixedPointGuess(const Vector3d& pelvis_position,
+                            const RigidBodyTree<double>& tree,
+                            VectorXd* q_init,
+                            VectorXd* u_init,
+                            VectorXd* lambda_init) {
+  int n_q = tree.get_num_positions();
+  int n_v = tree.get_num_velocities();
+  int n_u = tree.get_num_actuators();
+
+  int toe_left_idx = GetBodyIndexFromName(tree, "toe_left");
+  int toe_right_idx = GetBodyIndexFromName(tree, "toe_right");
+  Vector3d pt_front_contact(-0.0457, 0.112, 0);
+  Vector3d pt_rear_contact(0.088, 0, 0);
+  MatrixXd xa(3, 4);
+  xa.col(0) = pt_front_contact;
+  xa.col(1) = pt_rear_contact;
+  xa.col(2) = pt_front_contact;
+  xa.col(3) = pt_rear_contact;
+  std::vector<int> idxa;
+  idxa.push_back(toe_left_idx);
+  idxa.push_back(toe_left_idx);
+  idxa.push_back(toe_right_idx);
+  idxa.push_back(toe_right_idx);
+  ContactInfo contact_info(xa, idxa);
+
+  VectorXd q_desired = VectorXd::Zero(n_q);
+  q_desired << -0.0872062,
+            1.56883E-13,
+            1,
+            1,
+            1.9884E-13,
+            1.30167E-14,
+            -2.10728E-14,
+            -0.0104994,
+            0.0104994,
+            -3.42713E-07,
+            -3.30511E-07,
+            0.509601,
+            0.509602,
+            -1.2219,
+            -1.22191,
+            1.44602,
+            1.44602,
+            -1.6072,
+            -1.6072;
+
+  std::map<int, double> fixed_joints;
+  for (int i = 0; i < 3; i++) {
+    fixed_joints[i] = pelvis_position[i];
+  }
+  fixed_joints[3] = 1;
```

Comment:
Which joints are these?

Reply:
0 to 2 is base x y z, 3 to 6 is quaternion, and 9 and 10 is hip yaw. (documented in new commits)